Update dependencies.

XhmikosR opened the pull request:
I left workbox-build @2.1.3 since I can't find any docs what they have changed and with v3.0.0 it's broken.

Maybe @Lyricalz can help us out :)

TODO:

- [x] Use sinon from a CDN
- [x] Use QUnit from a CDN
- [x] Update workbox to v3.0.0

XhmikosR committed:
Update dependencies.

XhmikosR committed:
tests: use QUnit from the jQuery CDN.

XhmikosR committed:
Use sinon from cdnjs.

Johann-S commented:
They are a new release of Popper.js (released yesterday): 1.14.1

Johann-S commented on package.json:
```diff
@@ -95,38 +95,38 @@
     "shelljs": "^0.8.1",
     "shx": "^0.2.2",
-    "sinon": "^4.4.3",
+    "sinon": "^4.4.5",
```
should we upgrade to their new major release Sinon 5 ? Because currently we haven't unit tests which use Sinon (but it'll happen soon)

XhmikosR commented:
>There are a new release of Popper.js (released yesterday): 1.14.1

Yeah, but it doesn't have any changes to the dist files as far as I can tell.

As for sinon, I wasn't sure if v5 works so I stayed with v4 to be safe.

Johann-S commented:
We cannot update to QUnit 2.5.1 because of `karma-qunit` which seems dead... See: https://github.com/karma-runner/karma-qunit/issues/98

XhmikosR commented on js/tests/index.html:
```diff
@@ -26,7 +26,7 @@
 
     <!-- Sinon -->
-    <script src="https://cdnjs.cloudflare.com/ajax/libs/sinon.js/4.4.3/sinon-no-sourcemaps.min.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/sinon.js/4.4.5/sinon-no-sourcemaps.min.js"></script>
```
It's 404 that is why I haven't changed it. Please revert it.

Johann-S replied:
Indeed ! My bad

Lyricalz commented:
@XhmikosR I can make a PR this week?

XhmikosR commented:
@Lyricalz: sure thing, just CC me!

Johann-S commented:
Finally https://github.com/karma-runner/karma-qunit/issues/98 is merged, so we'll have to wait for their next release 👍

XhmikosR commented:
Cool, @Johann-S feel free to push here when done.

@Lyricalz: please target this branch in your workbox update PR so that we have everything in one place.

Johann-S commented:
I updated karma-qunit on my local branch and it doesn't work... So currently I don't understand where it failed in our karma plugins... Or if it's Karma itself

MikeyBeLike committed:
Update to workbox 3.0 & fix breaking changes

Lyricalz commented:
Made a PR for workbox here: https://github.com/twbs/bootstrap/pull/25873

XhmikosR committed:
sw.js: fix TypeError with new workbox.

XhmikosR committed:
Update workbox.js

XhmikosR committed:
Update all devDependencies.

Kept back karma-qunit and qunitjs because tests fail.

XhmikosR commented:
@Johann-S: it seems we are hitting https://github.com/rollup/rollup/wiki/Troubleshooting#name-is-not-exported-by-module

I also updated `karma-qunit` and switched to `qunit`.

I think after we fix this new error, only the CDN switch is left along with updating qunit in tests.

Johann-S commented:
I don't think it's related to Rollup or Babel Plugins helpers but more related to : `@babel/proposal-object-rest-spread` which allow us to use object spread operator (see: https://travis-ci.org/twbs/bootstrap/jobs/355916540#L629)

Johann-S commented:
Maybe `npm install --save-dev @babel/plugin-proposal-object-rest-spread` should help us 🤔

Johann-S committed:
fix rollup config

Johann-S commented:
I fixed the previous error but they are another one 😢

Johann-S commented:
If I set qunitjs to 2.4.1 and karma-qunit to 1.2.1 everything works...

XhmikosR commented:
I guess I will revert that but it would be nice if we could pinpoint the
cause and report it or fix it.

On Tue, Mar 20, 2018, 23:47 Johann-S <notifications@github.com> wrote:

> If I set qunitjs to 2.4.1 and karma-qunit to 1.2.1 everything works...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/twbs/bootstrap/pull/25840#issuecomment-374767288>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAVVtajArJ4-xcud1lMubt6kS4i30Sflks5tgXkAgaJpZM4SpGU5>
> .
>

Johann-S commented:
I checked a lot karma-qunit and for me it's more related to a change in QUnit which should be handled in karma-qunit

XhmikosR commented:
I'll try downgrading karma-qunit tomorrow, unless you beat me to it.

XhmikosR commented:
I downgraded karma-qunit and qunitjs and tests work again.

XhmikosR committed:
tests: load 3rd-party libs from the local node_modules folder.

This reduces duplication and the chances to forget to update something.

XhmikosR committed:
Remove the local jquery-1.9.1.min.js.
